nixos/modules/services/security/cfssl.nix

```diff
@@ -0,0 +1,205 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.cfssl;
+in {
+  options.services.cfssl = {
+    enable = mkEnableOption ''
+      Whether to enable the CFSSL CA api-server.
+    '';
+
+    dataDir = mkOption {
+      default = "/var/lib/cfssl";
+      type = types.str;
+      description = "cfssl work directory";
+    };
+
+    address = mkOption {
+      default = "127.0.0.1";
+      type = types.str;
+      description = "Address to bind";
+    };
+
+    port = mkOption {
+      default = 8888;
+      type = types.int;
```

Comment:
You could use the newish `types.ints.u16` to prevent users trying to assign an invalid port (have seen people do this)